apps/state/config/config.exs

```diff
@@ -25,7 +25,9 @@ config :state, :route_pattern,
     "Shuttle-AlewifeLittletonLocal-0-1" => false,
     # don't ignore Newton Connection RailBus for Worcester Line
     "Shuttle-NewtonHighlandsWellesleyFarms-0-0" => false,
-    "Shuttle-NewtonHighlandsWellesleyFarms-0-1" => false
+    "Shuttle-NewtonHighlandsWellesleyFarms-0-1" => false,
+    # don't ignore Providence trains stopping at Forest Hills
+    "CR-Providence-d01bc229-0" => false
```

Comment:
Hm, unfortunately this seems like it might be a little brittle, since the ID is liable to change, perhaps if a stop e.g. becomes a flag stop or a leave early. Is that likely?

When switching to these route pattern IDs, I changed this configuration list to use "prefixes", for the purposes of the `CR-Franklin-Foxboro-` at the top. But I don't think we can do `CR-Providence-` here, right? That would match too many.

Reply:
Depending how likely it is for `d01bc229` to change out from under us, we could extend the matching functionality a bit here to handle this CR format. I initially sketched out something that uses a regex, but unfortunately can't find a way to make that work cleanly, since this config is used at compile time to make macros, and you can't use a regex as a function guard. But we could have special handling for CR route patterns:

```ex
ignore_override_cr: %{
  {"CR-Providence-", "-0"} => false,
}
```

And then update the function generation code in `apps/lib/state/helpers.ex`:

```ex
for {{prefix, suffix}, ignore?} <-
      Application.get_env(:state, :route_patterns)[:ignore_override_cr] do
  def ignore_trip_route_pattern?(%Trip{route_pattern_id: unquote(prefix) <> binary-size(8) <> unquote(suffix)}), do: unquote(ignore?)
```

But I'm not sure how likely the CR route pattern is to change out from under us, if this is all worth it.

Reply:
Hmm, I'd say that it's _relatively_ unlikely that the route pattern would change, but it's certainly not impossible. If it was to happen, though, it would be during one of the (usually) semiannual major schedule changes, during which we make sure to test all new schedules on dev-green, with an eye toward peculiarities such as these late-night connecting trains. So my worries aren't too high on this.

Another solution could come from GTFS-creator: The existing prefix `CR-Franklin-Foxboro-` used to be generated by us upon detection of such a trip for this very purpose: see https://github.com/mbta/gtfs_creator/blob/136c04a28491d6f16fee69cce41fc35083388aa2/gtfs_creator2/commuter_rail/reader.py#L96. We could do a very similar thing to pick up this trip and give it a prefix of like `CR-Providence-ForestHills-`, and then intercept it here in the API's config. The con, of course, would be adding a bit more to GTFS-creator's existing bloat. 🙂

Reply:
Hmm, I see. Yeah, I don't want to mess with GTFS-creator right now for that, so I think this is fine for now.